src/workspaces/common/state/useCloningWorkspaceNotifications.test.tsx

```diff
@@ -0,0 +1,143 @@
+import { DeepPartial } from '@terra-ui-packages/core-utils';
+import { NotificationType } from '@terra-ui-packages/notifications';
+import { waitFor } from '@testing-library/react';
+import React from 'react';
+import { Ajax } from 'src/libs/ajax';
+import { clearNotification, notify } from 'src/libs/notifications';
+import { cloningWorkspacesStore } from 'src/libs/state';
+import { asMockedFn, renderWithAppContexts as render } from 'src/testing/test-utils';
+import { defaultAzureWorkspace } from 'src/testing/workspace-fixtures';
+import {
+  notifyNewWorkspaceClone,
+  useCloningWorkspaceNotifications,
+} from 'src/workspaces/common/state/useCloningWorkspaceNotifications';
+import { WorkspaceInfo, WorkspaceState } from 'src/workspaces/utils';
+
+type AjaxContract = ReturnType<typeof Ajax>;
+type AjaxWorkspacesContract = AjaxContract['Workspaces'];
+
+jest.mock('src/libs/ajax', (): typeof import('src/libs/ajax') => {
+  return {
+    ...jest.requireActual('src/libs/ajax'),
+    Ajax: jest.fn(),
+  };
+});
+
+type NotificationExports = typeof import('src/libs/notifications');
+jest.mock<NotificationExports>(
+  'src/libs/notifications',
+  (): NotificationExports => ({
+    ...jest.requireActual('src/libs/notifications'),
+    notify: jest.fn(),
+    clearNotification: jest.fn(),
+  })
+);
+// notify
+const CloningTestComponent = (): React.ReactNode => {
+  useCloningWorkspaceNotifications();
+  return null;
+};
+
+describe('useCloningWorkspaceNotifications', () => {
+  beforeEach(() => {
+    cloningWorkspacesStore.set([]);
+  });
+
+  describe('notifyNewWorkspaceClone', () => {
+    it('adds the workspace to the cloning store', () => {
+      // Arrange
+      const clone: WorkspaceInfo = {
+        ...defaultAzureWorkspace.workspace,
+        state: 'Cloning',
+      };
+      // Act
+      notifyNewWorkspaceClone(clone);
+
+      // Assert
+      expect(cloningWorkspacesStore.get()).toEqual([clone]);
+    });
+
+    it('creates a notification that the clone has started', () => {
+      // Arrange
+      const clone: WorkspaceInfo = {
+        ...defaultAzureWorkspace.workspace,
+        state: 'Cloning',
+      };
+      // Act
+      notifyNewWorkspaceClone(clone);
+
+      // Assert
+      expect(asMockedFn(notify)).toHaveBeenCalledWith('info', expect.any(Object), {
+        id: expect.any(String),
+        message: expect.any(Object),
+      });
+    });
+  });
+
+  beforeEach(() => {
```

Comment:
There is already a `beforeEach` above. Can you combine them?

Reply:
Oops - I thought the other one was inside the nesting.